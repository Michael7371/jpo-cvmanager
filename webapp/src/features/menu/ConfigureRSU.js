import { useState } from 'react'
import SnmpwalkMenu from '../../components/SnmpwalkMenu'
import SnmpsetMenu from '../../components/SnmpsetMenu'
import RsuRebootMenu from '../../components/RsuRebootMenu'
import Accordion from '@mui/material/Accordion'
import AccordionDetails from '@mui/material/AccordionDetails'
import AccordionSummary from '@mui/material/AccordionSummary'
import { useSelector } from 'react-redux'
import { ThemeProvider, createTheme } from '@mui/material'
import ExpandMoreIcon from '@mui/icons-material/ExpandMore'
import Typography from '@mui/material/Typography'
import { selectSelectedRsu } from '../../generalSlices/rsuSlice'

import '../../components/css/SnmpwalkMenu.css'

const ConfigureRSU = () => {
  const [expanded, setExpanded] = useState(false)
  const handleChange = (panel) => (event, isExpanded) => {
    setExpanded(isExpanded ? panel : false)
  }
  const selectedRsu = useSelector(selectSelectedRsu)

  return (
    <div>
      {selectedRsu ? (
        <div>
          <h2 className="snmpheader">RSU</h2>
          <h2 className="snmpheader2">
            Roadway: {selectedRsu.properties.primary_route}
            <br />
            Milepost: {String(selectedRsu.properties.milepost)}
            <br />
            IPv4: {selectedRsu.properties.ipv4_address}
          </h2>
        </div>
      ) : (
        <h2 className="snmpheader2">Select a RSU to configure on the Map tab</h2>
      )}
      <div id="sideBarBlock" className="accordion">
        <ThemeProvider theme={accordionTheme}>
          <Accordion className="accordion-content" expanded={expanded === 'panel1'} onChange={handleChange('panel1')}>
            <AccordionSummary
              expandIcon={<ExpandMoreIcon className="expand" />}
              aria-controls="panel1bh-content"
              id="panel1bh-header"
            >
              <Typography>Current Configuration</Typography>
            </AccordionSummary>
            <ThemeProvider theme={innerAccordionTheme}>
              <Accordion>
                <AccordionDetails>
                  <SnmpwalkMenu />
                </AccordionDetails>
              </Accordion>
            </ThemeProvider>
          </Accordion>
          <Accordion expanded={expanded === 'panel2'} onChange={handleChange('panel2')}>
            <AccordionSummary
              expandIcon={<ExpandMoreIcon className="expand" />}
              aria-controls="panel2bh-content"
              id="panel2bh-header"
              className="expand"
            >
              <Typography>Add Message Forwarding</Typography>
            </AccordionSummary>
            <ThemeProvider theme={innerAccordionTheme}>
              <Accordion>
                <AccordionDetails>
                  <SnmpsetMenu />
                </AccordionDetails>
              </Accordion>
            </ThemeProvider>
          </Accordion>
          <Accordion expanded={expanded === 'panel3'} onChange={handleChange('panel3')}>
            <AccordionSummary
              className="expand"
              expandIcon={<ExpandMoreIcon className="expand" />}
              aria-controls="panel3bh-content"
              id="panel3bh-header"
            >
              <Typography>Reboot</Typography>
            </AccordionSummary>
            <ThemeProvider theme={innerAccordionTheme}>
              <Accordion>
                <AccordionDetails>
                  <RsuRebootMenu />
                </AccordionDetails>
              </Accordion>
            </ThemeProvider>
          </Accordion>
        </ThemeProvider>
      </div>
    </div>
  )
}

const accordionTheme = createTheme({
  palette: {
    text: {
      primary: '#ffffff',
      secondary: '#ffffff',
      disabled: '#ffffff',
      hint: '#ffffff',
    },
    divider: '#333',
    background: {
      paper: '#333',
    },
  },
})
const innerAccordionTheme = createTheme({
  palette: {
    text: {
      primary: '#fff',
      secondary: '#fff',
      disabled: '#fff',
      hint: '#fff',
    },
    divider: '#333',
    background: {
      paper: '#333',
    },
  },
})

export default ConfigureRSU
