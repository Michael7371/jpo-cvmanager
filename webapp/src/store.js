import { configureStore } from '@reduxjs/toolkit'
import rsuReducer from './generalSlices/rsuSlice'
import userReducer from './generalSlices/userSlice'
import wzdxReducer from './generalSlices/wzdxSlice'
import configReducer from './generalSlices/configSlice'
import adminAddOrganizationReducer from './features/adminAddOrganization/adminAddOrganizationSlice'
import adminAddRsuReducer from './features/adminAddRsu/adminAddRsuSlice'
import adminAddUserReducer from './features/adminAdduser/adminAddUserSlice'
import adminEditOrganizationReducer from './features/adminEditOrganization/adminEditOrganizationSlice'
import adminEditRsuReducer from './features/adminEditRsu/adminEditRsuSlice'
import adminEditUserReducer from './features/adminEditUser/adminEditUserSlice'
import adminOrganizationTabReducer from './features/adminOrganizationTab/adminOrganizationTabSlice'
import adminOrganizationTabUserReducer from './features/adminOrganizationTabUser/adminOrganizationTabUserSlice'
import adminOrganizationTabRsuReducer from './features/adminOrganizationTabRsu/adminOrganizationTabRsuSlice'
import adminRsuTabReducer from './features/adminRsuTab/adminRsuTabSlice'
import adminUserTabReducer from './features/adminUserTab/adminUserTabSlice'
import menuReducer from './features/menu/menuSlice'
import rsuUpdateMenuReducer from './features/rsuUpdateMenu/rsuUpdateMenuSlice'

export const setupStore = (preloadedState) => {
  return configureStore({
    reducer: {
      rsu: rsuReducer,
      user: userReducer,
      wzdx: wzdxReducer,
      config: configReducer,
      adminAddOrganization: adminAddOrganizationReducer,
      adminAddRsu: adminAddRsuReducer,
      adminAddUser: adminAddUserReducer,
      adminEditOrganization: adminEditOrganizationReducer,
      adminEditRsu: adminEditRsuReducer,
      adminEditUser: adminEditUserReducer,
      adminOrganizationTab: adminOrganizationTabReducer,
      adminOrganizationTabUser: adminOrganizationTabUserReducer,
      adminOrganizationTabRsu: adminOrganizationTabRsuReducer,
      adminRsuTab: adminRsuTabReducer,
      adminUserTab: adminUserTabReducer,
      menu: menuReducer,
      rsuUpdateMenu: rsuUpdateMenuReducer,
    },
    preloadedState,
  })
}
