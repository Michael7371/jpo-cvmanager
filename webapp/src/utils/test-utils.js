/**
 * This function searches for the every somewhat random/chaotic class name and property that cause snapshot tests to be inconsistent.
 * This current list includes MUI classes (css-*) and aria-invalid attributes.
 *
 * @ex :
 * ```
 * const { container } = render(<Component />);
 *
 * replaceChaoticIds(container);
 * ```
 *
 * @param container The HTMLElement node to search for SSR ids
 */
function replaceChaoticIds(container) {
  const props = [
    {
      selector: 'class',
      updateFunc: (val) => val.replace(/css-[0-9a-z]*?-/g, 'css-mocked-'),
    },
  ]

  container.querySelectorAll('*').forEach((item) => {
    props.forEach((prop) => {
      if (item.getAttribute(prop.selector)) {
        item.setAttribute(prop.selector, prop.updateFunc(item.getAttribute(prop.selector)))
      }
    })
  })
  container.querySelectorAll('input[aria-invalid]').forEach((item) => {
    item.removeAttribute('aria-invalid')
  })
  return container
}

export { replaceChaoticIds }
