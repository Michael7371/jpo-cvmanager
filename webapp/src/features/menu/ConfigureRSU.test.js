import React from 'react'
import { render } from '@testing-library/react'
import ConfigureRSU from './ConfigureRSU'
import { Provider } from 'react-redux'
import { setupStore } from '../../store'
import { replaceChaoticIds } from '../../utils/test-utils'

it('should take a snapshot', () => {
  const { container } = render(
    <Provider store={setupStore({})}>
      <ConfigureRSU />
    </Provider>
  )

  expect(replaceChaoticIds(container)).toMatchSnapshot()
})
