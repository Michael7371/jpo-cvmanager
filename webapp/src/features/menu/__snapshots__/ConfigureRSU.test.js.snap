// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<div>
  <div>
    <h2
      class="snmpheader2"
    >
      Select a RSU to configure on the Map tab
    </h2>
    <div
      class="accordion"
      id="sideBarBlock"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters accordion-content css-mocked-MuiPaper-root-MuiAccordion-root"
      >
        <div
          aria-controls="panel1bh-content"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-mocked-MuiButtonBase-root-MuiAccordionSummary-root"
          id="panel1bh-header"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-mocked-MuiAccordionSummary-content"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
            >
              Current Configuration
            </p>
          </div>
          <div
            class="MuiAccordionSummary-expandIconWrapper css-mocked-MuiAccordionSummary-expandIconWrapper"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium expand css-mocked-MuiSvgIcon-root"
              data-testid="ExpandMoreIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-mocked-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-mocked-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-mocked-MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="panel1bh-header"
                class="MuiAccordion-region"
                id="panel1bh-content"
                role="region"
              >
                <div
                  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-mocked-MuiPaper-root-MuiAccordion-root"
                >
                  <div
                    class="MuiAccordionDetails-root css-mocked-MuiAccordionDetails-root"
                  >
                    <div
                      id="snmpdiv"
                    >
                      <h2
                        id="snmpheader"
                      >
                        Current Message Forwarding
                      </h2>
                      <div />
                      <div />
                    </div>
                  </div>
                  <div
                    class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-mocked-MuiCollapse-root"
                    style="min-height: 0px;"
                  >
                    <div
                      class="MuiCollapse-wrapper MuiCollapse-vertical css-mocked-MuiCollapse-wrapper"
                    >
                      <div
                        class="MuiCollapse-wrapperInner MuiCollapse-vertical css-mocked-MuiCollapse-wrapperInner"
                      >
                        <div
                          class="MuiAccordion-region"
                          role="region"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-mocked-MuiPaper-root-MuiAccordion-root"
      >
        <div
          aria-controls="panel2bh-content"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters expand css-mocked-MuiButtonBase-root-MuiAccordionSummary-root"
          id="panel2bh-header"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-mocked-MuiAccordionSummary-content"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
            >
              Add Message Forwarding
            </p>
          </div>
          <div
            class="MuiAccordionSummary-expandIconWrapper css-mocked-MuiAccordionSummary-expandIconWrapper"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium expand css-mocked-MuiSvgIcon-root"
              data-testid="ExpandMoreIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-mocked-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-mocked-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-mocked-MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="panel2bh-header"
                class="MuiAccordion-region"
                id="panel2bh-content"
                role="region"
              >
                <div
                  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-mocked-MuiPaper-root-MuiAccordion-root"
                >
                  <div
                    class="MuiAccordionDetails-root css-mocked-MuiAccordionDetails-root"
                  >
                    <div
                      id="snmpdiv"
                    >
                      <h2
                        id="snmpheader"
                      >
                        Add Message Forwarding
                      </h2>
                      <form
                        id="snmpform"
                      >
                        <label
                          id="snmplabel"
                        >
                          <strong>
                            Destination IP:
                          </strong>
                          <input
                            id="snmpinput"
                            type="text"
                            value=""
                          />
                        </label>
                        <label
                          id="snmplabel"
                        >
                          <strong>
                            Message Type:
                          </strong>
                          <select
                            id="snmpdropdown"
                          >
                            <option
                              value="bsm"
                            >
                              BSM
                            </option>
                            <option
                              value="spat"
                            >
                              SPaT
                            </option>
                            <option
                              value="map"
                            >
                              MAP
                            </option>
                            <option
                              value="srm"
                            >
                              SRM
                            </option>
                            <option
                              value="ssm"
                            >
                              SSM
                            </option>
                          </select>
                        </label>
                      </form>
                      <button
                        id="refreshbtn"
                      >
                        Submit Config
                      </button>
                      <p
                        id="infotext"
                      >
                        Only message type and index is required for delete
                      </p>
                      <div />
                      <div />
                      <div />
                    </div>
                  </div>
                  <div
                    class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-mocked-MuiCollapse-root"
                    style="min-height: 0px;"
                  >
                    <div
                      class="MuiCollapse-wrapper MuiCollapse-vertical css-mocked-MuiCollapse-wrapper"
                    >
                      <div
                        class="MuiCollapse-wrapperInner MuiCollapse-vertical css-mocked-MuiCollapse-wrapperInner"
                      >
                        <div
                          class="MuiAccordion-region"
                          role="region"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-mocked-MuiPaper-root-MuiAccordion-root"
      >
        <div
          aria-controls="panel3bh-content"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters expand css-mocked-MuiButtonBase-root-MuiAccordionSummary-root"
          id="panel3bh-header"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-mocked-MuiAccordionSummary-content"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
            >
              Reboot
            </p>
          </div>
          <div
            class="MuiAccordionSummary-expandIconWrapper css-mocked-MuiAccordionSummary-expandIconWrapper"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium expand css-mocked-MuiSvgIcon-root"
              data-testid="ExpandMoreIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-mocked-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-mocked-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-mocked-MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="panel3bh-header"
                class="MuiAccordion-region"
                id="panel3bh-content"
                role="region"
              >
                <div
                  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-mocked-MuiPaper-root-MuiAccordion-root"
                >
                  <div
                    class="MuiAccordionDetails-root css-mocked-MuiAccordionDetails-root"
                  >
                    <div
                      id="snmpdiv"
                    >
                      <h2
                        id="snmpheader"
                      >
                        Administrator Reboot
                      </h2>
                      <button
                        id="refreshbtn"
                      >
                        Reboot
                      </button>
                      <p
                        id="infotext"
                      >
                        Warning: This action could result in taking the RSU offline
                      </p>
                    </div>
                  </div>
                  <div
                    class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-mocked-MuiCollapse-root"
                    style="min-height: 0px;"
                  >
                    <div
                      class="MuiCollapse-wrapper MuiCollapse-vertical css-mocked-MuiCollapse-wrapper"
                    >
                      <div
                        class="MuiCollapse-wrapperInner MuiCollapse-vertical css-mocked-MuiCollapse-wrapperInner"
                      >
                        <div
                          class="MuiAccordion-region"
                          role="region"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
